src/main/java/racingcar/vehicle/Distance.java

```diff
@@ -0,0 +1,75 @@
+package racingcar.vehicle;
+
+import java.util.Objects;
+import racingcar.common.exception.InvalidInitDistanceException;
+import racingcar.common.exception.InvalidMoveDistanceException;
+
+public class Distance implements Comparable<Distance> {
+
+    private static final int MIN_INIT_INPUT = 0;
+    private static final int MIN_MOVE_INPUT = 0;
+
+    private final int distance;
+
+    public Distance(int distance) {
+        if (!initValidation(distance)) {
+            throw new InvalidInitDistanceException();
+        }
+
+        this.distance = distance;
+    }
+
+    public Distance move(int moveDistance) {
+        if (!moveValidation(moveDistance)) {
+            throw new InvalidMoveDistanceException();
+        }
+
+        return new Distance(this.get() + moveDistance);
+    }
```

Comment:
비즈니스 요구사항을 풀어내는 경우에는 getter를 최대한 지양하는 것이 좋습니다. 

이 경우는 같은 클래스의 객체들이기 때문에 필드에 바로 접근이 가능합니다. 

```suggestion
        return new Distance(this.distance + moveDistance);
    }
```